src/Lists.js

```diff
@@ -0,0 +1,264 @@
+import { setDoc, doc } from "firebase/firestore";
+import { query, collection } from "firebase/firestore";
+import { useCollectionData } from "react-firebase-hooks/firestore";
+import Tab from '@mui/material/Tab';
+import Tabs from '@mui/material/Tabs';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import Typography from '@mui/material/Typography';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import MenuIcon from '@mui/icons-material/Menu';
+import IconButton from '@mui/material/IconButton';
+import List from './List';
+
+// web app's Firebase configuration
+const firebaseConfig = {
+    apiKey: "AIzaSyDnLHzrDzijh4KmeJWRU5zSIiW2cPsZRHU",
+    authDomain: "cs124-lab3-3028f.firebaseapp.com",
+    projectId: "cs124-lab3-3028f",
+    storageBucket: "cs124-lab3-3028f.appspot.com",
+    messagingSenderId: "426502461839",
+    appId: "1:426502461839:web:56b4c42f33c8a6187353fa"
+};
+    
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+const db = getFirestore(app);
+
+export default function Lists(props) {
+    const q = query(collection(db, 'Lists'));   
+    const [lists, loading, error] = useCollectionData(q);
+    const [tabId, setTabId] = useState((lists && lists.length!==0) ? lists[0].id : 'none');
+
+    const isNarrowThan230 = useMediaQuery({ maxWidth: 230 })
+    const ITEM_HEIGHT = 48;
+    const [anchorElMenu, setAnchorElMenu] = React.useState(null);
+    const openMenu = Boolean(anchorElMenu);
+
+    const [newListDialogOpen, setNewListDialogOpen] = React.useState(false);
+    const [listName, setListName] = useState("");
+    const [showAlert, setShowAlert] = React.useState(false);
+    
+    const handleChangeTab = (event, newId) => {
+        setTabId(newId);
+    };
+
+    const handleNewListDialogOpen = () => {
+        setNewListDialogOpen(true);
+    };
+
+    const handleNewListDialogClose = () => {
+        setNewListDialogOpen(false);
+        setShowAlert(false);
+        setListName("")
+    };
+
+    const handleSetListName = e => {
+        setListName(e.target.value)
+        if (e.target.value.length>0) {
+            setShowAlert(false);
+        }else{
+            setShowAlert(true);
+        }
+    }
+
+    function submitNewList(){
+        const listId = generateUniqueID()
+        setDoc(doc(db, "Lists", listId), {
+            id: listId,
+            name: listName,
+            showCompleted: true,
+            sortBy: "priority",
+            sortDir: "desc",
+            created: serverTimestamp(),
+        })
+        if (lists.length===0) {
```

Comment:
Not the right place to do this.  If lists.length is 0 coming into this function, it'll be 0 here, too.